src/wallet/wallet2.cpp

```diff
@@ -1459,24 +1460,30 @@ void wallet2::update_pool_state()
   MDEBUG("update_pool_state done first loop");
 
   // remove pool txes to us that aren't in the pool anymore
-  std::unordered_map<crypto::hash, wallet2::payment_details>::iterator uit = m_unconfirmed_payments.begin();
-  while (uit != m_unconfirmed_payments.end())
+  // but only if we just refreshed, so that the tx can go in
+  // the in transfers list instead (or nowhere if it just
+  // disappeared without being mined)
+  if (refreshed)
   {
-    const crypto::hash &txid = uit->second.m_tx_hash;
-    bool found = false;
-    for (const auto &it2: res.tx_hashes)
+    std::unordered_map<crypto::hash, wallet2::payment_details>::iterator uit = m_unconfirmed_payments.begin();
+    while (uit != m_unconfirmed_payments.end())
```

Comment:
I'm not sure how that worked before... Overhead isn't really relevant as this is a mostly empty map. Can you expand on what you're asking in the third sentence ?